tests/std/tests/P0896R4_views_take_while/test.cpp

```diff
@@ -0,0 +1,493 @@
+        STATIC_ASSERT(noexcept(move(as_const(rng)) | take_while_even) == is_noexcept);
+
+        STATIC_ASSERT(same_as<decltype(move(as_const(rng)) | pipeline), pipeline_t<const remove_reference_t<Rng>>>);
+        STATIC_ASSERT(noexcept(move(as_const(rng)) | pipeline) == is_noexcept);
+    } else if constexpr (!is_view && enable_borrowed_range<remove_cvref_t<Rng>>) {
+        using S                    = decltype(ranges::subrange{move(as_const(rng))});
+        using RS                   = take_while_view<S, Pred>;
+        constexpr bool is_noexcept = noexcept(S{move(as_const(rng))});
+
+        STATIC_ASSERT(same_as<decltype(views::take_while(move(as_const(rng)), is_less_than_three)), RS>);
+        STATIC_ASSERT(noexcept(views::take_while(move(as_const(rng)), is_less_than_three)) == is_noexcept);
+
+        STATIC_ASSERT(same_as<decltype(move(as_const(rng)) | take_while_even), RS>);
+        STATIC_ASSERT(noexcept(move(as_const(rng)) | take_while_even) == is_noexcept);
+
+        STATIC_ASSERT(same_as<decltype(move(as_const(rng)) | pipeline), pipeline_t<const remove_reference_t<Rng>>>);
+        STATIC_ASSERT(noexcept(move(as_const(rng)) | pipeline) == is_noexcept);
+    }
+
+    // Validate deduction guide
+#if !defined(__clang__) && !defined(__EDG__) // TRANSITION, DevCom-1159442
+    (void) 42;
+#endif // TRANSITION, DevCom-1159442
+    same_as<R> auto r = take_while_view{forward<Rng>(rng), is_less_than_three};
+    assert(ranges::equal(r, expected));
+
+    { // Validate take_while_view::pred
+        [[maybe_unused]] same_as<Pred> auto pred_copy = as_const(r).pred();
+        STATIC_ASSERT(noexcept(as_const(r).pred()));
+    }
+
+    const bool is_empty = ranges::empty(expected);
+
+    // Validate view_interface::empty and operator bool
+    STATIC_ASSERT(CanMemberEmpty<R> == forward_range<Rng>);
+    STATIC_ASSERT(CanBool<R> == CanEmpty<R>);
+    if constexpr (CanMemberEmpty<R>) {
+        assert(r.empty() == is_empty);
+        assert(static_cast<bool>(r) == !is_empty);
+    } else {
+        STATIC_ASSERT(CanEmpty<R> == CanSize<R>);
+        if constexpr (CanEmpty<R>) {
+            assert(ranges::empty(r) == is_empty);
+            assert(static_cast<bool>(r) == !is_empty);
+        }
+    }
+
+    STATIC_ASSERT(CanMemberEmpty<const R> == forward_range<const Rng>);
+    STATIC_ASSERT(CanBool<const R> == CanEmpty<const R>);
+    if constexpr (CanMemberEmpty<const R>) {
+        assert(as_const(r).empty() == is_empty);
+        assert(static_cast<bool>(as_const(r)) == !is_empty);
+    } else {
+        STATIC_ASSERT(CanEmpty<const R> == CanSize<const R>);
+        if constexpr (CanEmpty<const R>) {
+            assert(ranges::empty(as_const(r)) == is_empty);
+            assert(static_cast<bool>(as_const(r)) == !is_empty);
+        }
+    }
+
+    // Validate take_while_view::begin
+    STATIC_ASSERT(CanMemberBegin<R>);
+    STATIC_ASSERT(CanMemberBegin<const R>);

```

Comment:
This should only hold when `range<const R>` is `true`, which tells me that we're missing coverage adapting mutable-only ranges.

Reply:
Urgh, you know sometimes I just lack the imagination to create those evil things